src/util/time_tricks.rs

```diff
@@ -63,16 +91,52 @@ impl InstantExt for Instant {
             .expect("clock to go forwards from unix epoch")
     }
 
+    fn from_unix_duration(d: Duration) -> Self {
+        let since_beginning_of_time = d.saturating_sub(beginning_to_epoch());
+        BEGINNING_OF_TIME.0 + since_beginning_of_time
+    }
+
     fn to_ntp_duration(&self) -> Duration {
-        // RTP spec "wallclock" uses NTP time, which starts at 1900-01-01.
-        //
-        // https://tools.ietf.org/html/rfc868
-        //
-        // 365 days * 70 years + 17 leap year days
-        // (365 * 70 + 17) * 86400 = 2208988800
-        const MICROS_1900: Duration = Duration::from_micros(2_208_988_800 * 1_000_000 as u64);
-
-        self.to_unix_duration() + MICROS_1900
+        self.to_unix_duration() + Duration::from_micros(MICROS_1900)
+    }
+
+    fn from_ntp_64(v: u64) -> Self {
+        // https://tools.ietf.org/html/rfc3550#section-4
+        // Wallclock time (absolute date and time) is represented using the
+        // timestamp format of the Network Time Protocol (NTP), which is in
+        // seconds relative to 0h UTC on 1 January 1900 [4]. The full
+        // resolution NTP timestamp is a 64-bit unsigned fixed-point number with
+        // the integer part in the first 32 bits and the fractional part in the
+        // last 32 bits.
+        let secs_ntp = (v as f64) / F32;
+
+        // Shift to UNIX EPOCH
+        let secs_epoch = secs_ntp - SECS_1900 as f64;
```

Comment:
This is the code path of user provided values and can panic because of overflow.For example, a sender can send an SR with the second part of the NTP timestamps set to `2_208_988_700` which will result in `2_208_988_700 - 2_208_988_800`, this will then panic on line 119 below

Reply:
Ok. Saturating it is.